Sources/Extensions/UIKit/UITableViewExtensions.swift

```diff
@@ -181,6 +181,18 @@ public extension UITableView {
         return indexPath.section < self.numberOfSections && indexPath.row < self.numberOfRows(inSection: indexPath.section)
     }
 
+    /// SwifterSwift: Safely scroll to possibly invalid IndexPath
+    ///
+    /// - Parameters:
+    ///   - indexPath: Target IndexPath to scroll to
+    ///   - scrollPosition: Scroll position
+    ///   - animated: Whether to animate or not
+    public func safeScrollToRow(at indexPath: IndexPath, at scrollPosition: UITableViewScrollPosition, animated: Bool) {
+        if indexPath.section < numberOfSections && indexPath.row < numberOfRows(inSection: indexPath.section) {
```

Comment:
Why not use `if isValidIndexPath(indexPath)`?

Reply:
https://github.com/SwifterSwift/SwifterSwift/pull/445#discussion_r181182596

Reply:
I don't understand your comment. `isValidIndexPath` is 10 lines above...

Reply:
Each extension should be able to be copied and pasted without a dependency on other extensions

Reply:
`isValidIndexPath` is part of the same extension. More so, it's part of the same file. It's literally the function above.

Reply:
Sorry, sometimes I do a poor job of explaining things. We don't allow using a SwifterSwift extension inside another SwifterSwift extension

Reply:
No @SD10 I think you explained it very well 🙂, @guykogus here is a further explanation:

```swift
extension A {
   func b() { ... }
   func c() { ... }
}

extension B {
   func b() { ... }
   func c() { ... }
}
```

- units from class `A` (`b` and `c`) **CAN NOT** be used in any unit for extension `B` and viceversa.
- unit `b` in class `A` **CAN NOT** be used in unit `c` in the same class extension
- all units in any extension should be **SELF DEPENDENT**, that means they **CAN NOT** rely on any other SwifterSwift unit.
- units might use other **system wide** units; example: I can use `UIColor` or `Int` inside an extension for `UILabel`, since we asssume these types are available togehter when I import `UIKit` into my project.

Reply:
@guykogus, here is how we will add this extension in SwifterSwift

```swift
extension UITableView {

	public func isValidIndexPath(_ indexPath: IndexPath) -> Bool {
		return indexPath.section < self.numberOfSections && indexPath.row < self.numberOfRows(inSection: indexPath.section)
	}

	public func safeScrollToRow(at indexPath: IndexPath, at scrollPosition: UITableViewScrollPosition, animated: Bool) {
		guard indexPath.section < self.numberOfSections else { return }
		guard indexPath.row < numberOfRows(inSection: indexPath.section) else { return }
		scrollToRow(at: indexPath, at: scrollPosition, animated: animated)
	}

}
```

Reply:
@guykogus, absolutely, that is how we all do software, no one likes to write the same code in 2+ places, this is common sense!, I see your point 💯

However, we see this project as a unique case where each extension should be encapsulated, even if this lead to some duplicated code.

Reply:
In this situation it seems pretty unnecessary, but ok.

Reply:
Hi guys, sorry kinda late to this interesting discussion.
I don't really mind if I have to rewrite it to meet the requirements of this project.

Reply:
Thank you @setoelkahfi 🤗 
Could you please replace the `if` condition with 2 `guard` statements like above

Reply:
@setoelkahfi added below, sorry for the delay, time zone differences 😅